Show on App Launch: Open launch option when set

mikescamell opened the pull request:
Task/Issue URL: https://app.asana.com/0/1207908166761516/1208167186867417/f 

### Description

Adds functionality to open the show on app launch option based on what has been set. 

If there is no specialised launch happening then we check the show on app launch setting:

1. If LastOpenedTab is set, we do nothing as this is current behaviour
2. If NewTabPage is set then we open a new tab
3. If SpecificPage is set we compare the url to the current selected tab, if it's not the same we open a new tab otherwise we do nothing 

Also as part of this PR, we add validation to the URL:

1. It is blank will set it as a default to duckduckgo.com
2. If it has no scheme will add http (in case the website does not currently support https)
3. Otherwise return the url what was passed in

### Steps to test this PR

_Last Opened Tab_
- [x] Set "Show on App Launch" to **Last Opened Tab** 
- [x] Exit Settings
- [x] Remember your current tab and tab count
- [x] Force close the app via recents
- [x] Open the app
- [x] Check the tab is the same as when you closed
- [x] Ensure the tab count has not increased

_New Tab Page_
- [x] Set "Show on App Launch" to **New Tab Page**
- [x] Exit Settings
- [x] Remember your current tab and tab count. If this is the **New Tab Page** then please open a website e.g. “bbc.com"
- [x] Force close the app via recents
- [x] Open the app
- [x] Check the tab is now the **New Tab Page**
- [x] Ensure the tab count **has** increased by one
- [x] Force close the app via recents
- [x] Open the app
- [x] Check the tab is still the **New Tab Page**
- [x] Ensure the tab count has **not** increased by one

_Specific Page_
- [x] Set "Show on App Launch" to **Specific Page**
- [x] Check the default URL is “https://duckduckgo.com/” 
- [x] Exit Settings
- [x] Remember your current tab (it should not be “https://duckduckgo.com/”) and tab count
- [x] Force close the app via recents
- [x] Open the app
- [x] Check the tab is now **https://duckduckgo.com/**
- [x] Ensure the tab count **has** increased by one (if your tab was on the new tab page before you force closed it would not increase)
- [x] Force close the app via recents
- [x] Open the app
- [x] Check the tab is still **https://duckduckgo.com/**
- [ ] Ensure the tab count has **not** increased by one

_Specific Page Lowercasing_
- [x] Set "Show on App Launch" to **Specific Page**
- [x] Change the default URL to “BBC.com”
- [x] Press back
- [x] Check that under the "Show on App Launch” setting the url is “http://bbc.com”

_Specific Page Updated_
- [x] Set "Show on App Launch" to **Specific Page**
- [x] Change the default URL to a different url e.g. “cnn.com"
- [x] Exit Settings
- [x] Remember your current tab (it should **not** be the same url you entered earlier) and tab count
- [x] Force close the app via recents
- [x] Open the app
- [x] Check the tab is now the url you set e.g. **cnn.com**
- [x] Ensure the tab count **has** increased by one

_Specific Page Not a URL_
- [x] Set "Show on App Launch" to **Specific Page**
- [x] Change the default URL to a url that is invalid e.g. “example"
- [x] Press back
- [x] Check that under the "Show on App Launch” setting the url is “http://example”
- [x] Remember your current tab (it should **not** be the same url you entered earlier) and tab count
- [x] Force close the app via recents
- [x] Open the app
- [x] Check the tab has attempted to load "http://example"
- [x] Ensure the tab count **has** increased by one

_Specific Page Different Scheme_
- [x] Set "Show on App Launch" to **Specific Page**
- [x] Change the default URL to a url with a different scheme e.g. “ftp://file.com"
- [x] Press back
- [x] Check that under the "Show on App Launch” setting the url is still “ftp://file.com”

### UI changes

N/A

### Demo

https://github.com/user-attachments/assets/6296bd2e-d083-46fd-a738-787e8bf2b6ee

mikescamell commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4995?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4995** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4995?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#4980** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4980?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4952** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4952?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4950** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4950?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4946** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4946?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#4944** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4944?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @mikescamell and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

anikiki reviewed (approved):
LGTM :+1: 

I didn't tick `Ensure the tab count has *not increased by one` as the tab count is increased due to the trailing "/".

mikescamell commented:
> LGTM 👍
> 
> I didn't tick `Ensure the tab count has *not increased by one` as the tab count is increased due to the trailing "/".

I’ve moved the default url to be `https://duckduckgo.com/` seeing as we always append the slash. There’s no good reason for that not to be the default. I updated the test cases to reflect this.
